paymentsheet/src/main/java/com/stripe/android/paymentsheet/specifications/BankRepository.kt

```diff
@@ -0,0 +1,43 @@
+package com.stripe.android.paymentsheet.specifications
+
+import android.content.res.Resources
+import androidx.annotation.VisibleForTesting
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import java.io.InputStream
+
+internal data class BankRepository(
+    val resources: Resources
+) {
+    private val bankItemMap = mutableMapOf<SupportedBankType, List<DropdownItem>?>()
+
+    internal fun get(bankType: SupportedBankType) = requireNotNull(bankItemMap[bankType])
+
+    fun init() {
+        init(
+            SupportedBankType.values().associateWith { bank ->
+                val assets = resources.assets
+                assets.open(bank.assetFileName)
+            }
+        )
+    }
```

Comment:
Can these `init` methods be moved into the object initialization, with the overloaded one below being an optional constructor parameter? Same for `ResourceRepository` and `AddressFieldElementRepository`

Reply:
Because there is IO happening here and I am using dagger for injection, I have these init functions.   This is so that we can explicitly do the IO on a background thread.    This is happening [here](https://github.com/stripe/stripe-android/blob/f0fb7996c5db7bfc8efc4177d7b9049f111637cc/paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/Form.kt#L252)